airbyte-cdk/python/airbyte_cdk/sources/declarative/interpolation/macros.py

```diff
@@ -116,7 +123,7 @@ def format_datetime(dt: Union[str, datetime.datetime], format: str):
     """
     if isinstance(dt, datetime.datetime):
         return dt.strftime(format)
-    return parser.parse(dt).strftime(format)
+    return _str_to_datetime(dt).strftime(format)
```

Comment:
use `isoparse` instead of `parse` because the valid formats are more predictable.